File: src/0412-Fizz-Buzz/fizz_buzz.py
from typing import List

class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        output: List[str] = []
        for i in range(1, n+1):
            if i % 3 == 0:
                if i % 5 == 0:
                    output.append("FizzBuzz")
                else:
                    output.append("Fizz")
            elif i % 5 == 0:
                output.append("Buzz")
            else:
                output.append(f"{i}")
        return output
    
if __name__ == "__main__":
    n: int = 15
    solution = Solution()
    print(solution.fizzBuzz(n))

File: src/0412-Fizz-Buzz/go/fizz_buzz.go
package main

import (
	"fmt"
	"strconv"
)

func fizzBuzz(n int) []string {
	var res []string
	for i := 1; i <= n; i++ {
		if i%5 == 0 && i%3 == 0 {
			res = append(res, "FizzBuzz")
		} else if i%5 == 0 {
			res = append(res, "Buzz")
		} else if i%3 == 0 {
			res = append(res, "Fizz")
		} else {
			res = append(res, strconv.Itoa(i))
		}
	}

	return res
}

func main() {
	n := 3
	res := fizzBuzz(n)

	fmt.Println(res)
}
